nanocoat/lib/scritchui/scritchPencilRasterShapes.c

```diff
@@ -175,6 +818,94 @@ sjme_errorCode sjme_scritchpen_core_fillRect(
 	return sjme_error_default(error);
 }
 
+sjme_errorCode sjme_scritchpen_core_fillRoundRect(
+	sjme_attrInNotNull sjme_scritchui_pencil g,
+	sjme_attrInValue sjme_jint x,
+	sjme_attrInValue sjme_jint y,
+	sjme_attrInPositive sjme_jint w,
+	sjme_attrInPositive sjme_jint h,
+	sjme_attrInPositive sjme_jint arcWidth,
+	sjme_attrInPositive sjme_jint arcHeight)
+{
+	sjme_errorCode error;
+	sjme_scritchui_pencilFillRectFunc fillRect;
+	sjme_scritchui_pencilFillArcFunc fillArc;
+	sjme_jint xw, yh, arcwby2, archby2;
+
+	if (g == NULL)
+		return SJME_ERROR_NULL_ARGUMENTS;
+		
+	/* Lock. */
+	if (sjme_error_is(error = sjme_scritchpen_core_lock(g)))
+		return sjme_error_default(error);
+	
+	arcWidth = fabs(arcWidth);
+	arcHeight = fabs(arcHeight);
+
+	/**
+	 * We'll be doing only even arc widths and heights, otherwise the borders
+	 * will look off due to fractional rounding (java's AWT Graphics do allow
+	 * for odd width/heights though)
+	 */
+	if(arcWidth  %2 != 0) { arcWidth++; }
+	if(arcHeight %2 != 0) { arcHeight++; }
+	
+	/* The arcs cannot be larger than the rect's width/height*/
+	if(arcWidth >= w) { arcWidth = w-1; }
+	if(arcHeight >= h) { arcHeight = h-1; }
```

Comment:
Code style, single line statements should be on the following line and there should be no extra `{}`.